source/features/tag-changelog-link.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+This feature adds a compare link on each releases/tags/single tag page so that you can see
+what has changed since the previous release. If the tags are namespaced then it tries to
+get the previous release of the same namespaced tag.
+
+See it in action at: https://github.com/parcel-bundler/parcel/releases
+*/
+
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import fetchDom from '../libs/fetch-dom';
+import {diff} from '../libs/icons';
+import {isSingleTagPage} from '../libs/page-detect';
+import {getRepoPath, getRepoURL} from '../libs/utils';
+
+async function getNextPage(): Promise<DocumentFragment> {
+	const nextPageLink = select<HTMLAnchorElement>('.pagination a:last-child');
+	if (nextPageLink) {
+		return fetchDom(nextPageLink.href);
+	}
+
+	if (isSingleTagPage()) {
+		const [, tag = ''] = getRepoPath()!.split('releases/tag/', 2); // Already URL-encoded
+		return fetchDom(`/${getRepoURL()}/tags?after=${tag}`);
+	}
+
+	return new DocumentFragment();
+}
+
+async function init(): Promise<void | false> {
+	if (select.exists('.blankslate')) {
+		return false;
+	}
+
+	const documents = [document, await getNextPage()] as any; // TODO: fix select-dom types to accept mixed arrays
+	// These selectors need to work on:
+	// https://github.com/facebook/react/tags (tags list)
+	// https://github.com/facebook/react/releases (releases list)
+	// https://github.com/parcel-bundler/parcel/releases (releases list without release notes)
+	const tagElements = select.all('.f1 [href*="/releases/tag"], .commit-title [href*="/releases/tag"]', documents);
+	const commitElements = select.all('.muted-link[href*="/commit/"]', documents);
```

Comment:
This selector is not reliable. It is selecting multiple links on the single tag release page causing the wrong tag to be displayed as the previous tag. We'll need to change it.

Reply:
The same issue is also occurring on the release page where the tag is an actual release with the release notes. 

Apart from that, The feature has broken on such releases page due to index out of bounds. 

![Screen Shot 2019-05-03 at 12 18 45 AM](https://user-images.githubusercontent.com/22439276/57099248-9876d980-6d39-11e9-86a9-d5dcbe9e0998.png)